tests/ci/run_benchmark_build_tests.sh

```diff
@@ -82,7 +82,7 @@ build_openssl_no_debug $openssl_3_2_branch
 build_openssl_no_debug $openssl_master_branch
 build_boringssl
 
-run_build -DASAN=1 -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_CXX_STANDARD=14 -DCMAKE_C_STANDARD=11 -DENABLE_DILITHIUM=ON -DBENCHMARK_LIBS="\
+run_build -DASAN=1 -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_CXX_STANDARD=17 -DCMAKE_C_STANDARD=11 -DENABLE_DILITHIUM=ON -DBENCHMARK_LIBS="\
```

Comment:
NP: Unrelated change? Why?

Reply:
Had to cherry-pick https://github.com/aws/aws-lc/pull/2089/commits/b549c4a92f66f2fe422b9b6ac93fcda7f1706d0b
Testing was borked without.